### PR TITLE
Add guard for named tensors in the JIT

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25345 Add guards for using named tensor with serialization and multiprocessing
* **#25344 Add guard for named tensors in the JIT**

Test Plan
- [namedtensor ci]

Differential Revision: [D17101487](https://our.internmc.facebook.com/intern/diff/D17101487)

### CODE REVIEW

**Review by nairbv — APPROVED** [review]
seems reasonable to me but might want to wait for @eellison to approve.

Will jit eventually support named tensors?

**Review by eellison — APPROVED** [review]
LGTM. 

Note that this only checks border between python and JIT, so if someone uses Named Tensors in the middle of a script method this won't get hit. But there isn't really a better option, and this is also what we did for sparse tensors.

File: test/test_namedtensor.py
Comment (eellison) [reply test/test_namedtensor.py]: Yea returning a named tensor into python from script.

Comment (zou3519) [reply test/test_namedtensor.py]: Cool, I assumed that and added a test before I landed.

**Pull request closed without merging**
